yarn-project/accounts/src/testing/create_account.ts

```diff
@@ -12,24 +11,35 @@ import { getSchnorrAccount } from '../schnorr/index.js';
  */
 export function createAccount(pxe: PXE): Promise<AccountWalletWithSecretKey> {
   const secretKey = Fr.random();
-  const signingKey = sha512ToGrumpkinScalar([secretKey, GeneratorIndex.IVSK_M]);
+  const signingKey = deriveSigningKey(secretKey);
   return getSchnorrAccount(pxe, secretKey, signingKey).waitSetup();
 }
 
 /**
  * Creates a given number of random accounts using the Schnorr account wallet.
  * @param pxe - PXE.
  * @param numberOfAccounts - How many accounts to create.
+ * @param secrets - Optional array of secrets to use for the accounts. If empty, random secrets will be generated.
+ * @throws If the secrets array is not empty and does not have the same length as the number of accounts.
  * @returns The created account wallets.
  */
-export async function createAccounts(pxe: PXE, numberOfAccounts = 1): Promise<AccountWalletWithSecretKey[]> {
+export async function createAccounts(
+  pxe: PXE,
+  numberOfAccounts = 1,
+  secrets: Fr[] = [],
+): Promise<AccountWalletWithSecretKey[]> {
```

Comment:
Added secrets param here to be able to generate nsk_app for the account (I need the secret for that). There is a getNullifierKeys API on PXE database which is used by oracles but it didn't seem to make sense to expose it on PXE as I don't think it will be needed by anything else but this 1 test case.